components/higher-order/if-props-verify/index.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * WordPress dependencies
+ */
+import { getWrapperDisplayName } from '@wordpress/element';
+
+/**
+ * A Higher Order Component that provides a way to conditionally
+ * render a component depending on its props.
+ *
+ * @param {Function} predicate A predicate that receives the props passed to the component.
+ *                             Returns true if the component should be rendered and false otherwise.
+ *
+ * @returns {Component} Wrapped component.
+ */
+function ifPropsVerify( predicate ) {
+	return ( OriginalComponent ) => {
+		const WrappedComponent = ( props ) => predicate && predicate( props ) && <OriginalComponent { ...props } />;
```

Comment:
Mixed messaging here: We don't trust the developer to pass a function as `predicate`, but we do trust them to not return `0` as the result of `predicate` (I'll leave it to you to experiment with the outcome 😄 ).

Hint: https://codepen.io/aduth/pen/dWwzrL

Reply:
So the alternative here would be:

```jsx
predicate && predicate( props ) ? <OriginalComponent { ...props } /> : null
```

Reply:
or `!! predicate && !! predicate( props ) ? <OriginalComponent { ...props } />` depends if you like the `!!` or not